auslib/web/admin/swagger/api.yaml

```diff
@@ -108,6 +131,110 @@ parameters:
+    in: body
+    description: Request body containing options and data version field values for manipulating user permission values in DB
+    required: true
+    schema:
+      allOf:
+        - $ref: '#/definitions/CSRFModel'
+        - type: object
+          properties:
+            data_version:
+              type: integer
+              format: int32
+              description: data_version of permission
+              minimum: 1
+              example: 1
+            options:
+              type: ["string", "null"]
+              format: JSONStringField
+              description: JSON field specifying the actions that user can perform and the prdouct on which it has current permission to do so
+              example: "{'actions':['modify']}"
+          # accomodate 'options_as_json' and 'permission' which are duplicate values of 'options'
+          # and 'permission path parameter' respectively, getting passed in request body
+          # No need to validate the duplicate values again
+          additionalProperties: true
+
+
+responses:
+  updateExistingPermission:
+    description: "If the user already has the given permission, update its options and the data_version"
+    schema:
+      type: object
+      required:
+        - new_data_version
+      properties:
+        new_data_version:
+          description: "data version of the updated permission"
+          type: integer
+          format: int32
+          minimum: 2
+          example: 2
+
+  newObjectDataVersion:
+    description: "When a new object (permission/role) is successfully assigned to the user. Its data version value is always 1"
+    schema:
+      type: object
+      required:
+        - new_data_version
+      properties:
+        new_data_version:
+          description: "data version of the new object (permission/role) assigned.Value is always 1"
+          type: integer
+          format: int32
+          minimum: 1
+          maximum: 1
+          example: 1
+
+  invalidFormData:
+    description: Error in form data
+    schema:
+      $ref: '#/definitions/4xxClientErrorResponseModel'
+    examples:
+      application/json:
+        detail: "Invalid field value"
+        status: 400
+
+  unauthorizedUser:
+    description: Not authorized to access resource
+    schema:
+      $ref: "#/definitions/4xxClientErrorResponseModel"
+    examples:
+      application/json:
+        detail: "You are not authorized to access this resource"
+        status: 403
+
+  resourceNotFound:
+    description: Resource Not Found
+    schema:
+      $ref: "#/definitions/4xxClientErrorResponseModel"
+    examples:
+      application/json:
+        detail: "Requested resource was not found"
+        status: 404
+
+  defaultResponse:
+    description: Default case for sporadic exceptions. Example, No remote username and password in basic authentication for posting new data.
+    examples:
+      application/json:
+        Error: "REMOTE_USERNAME required"
```

Comment:
As per last week's discussion, there's no need to define the 'default' case at all. Removing all the default responses